Resync web-platform-tests/screen-orientation from upstream

marcoscaceres opened the pull request:
#### 04aa3aef7ee6095372062da56f660e966f1bf465
<pre>
Resync web-platform-tests/screen-orientation from upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=247111">https://bugs.webkit.org/show_bug.cgi?id=247111</a>
rdar://101619487

Reviewed by Youenn Fablet.

* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/active-lock-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/active-lock.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/event-before-promise-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/event-before-promise.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/fullscreen-interactions-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/fullscreen-interactions.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-basic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-basic.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-sandboxed-iframe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-sandboxed-iframe.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-unlock-check-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-unlock-check.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/nested-documents-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/nested-documents.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/non-fully-active.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/onchange-event-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/onchange-event-subframe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/onchange-event-subframe.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/onchange-event.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/orientation-reading-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/orientation-reading.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/resources/orientation-utils.js:
(export.async attachIframe):
(export.async loadIframe): Deleted.
(export.getOppositeOrientation): Deleted.
(export.makeCleanup.return.async if): Deleted.
(export.makeCleanup): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/resources/sandboxed-iframe-locking.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/unlock-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/unlock.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/active-lock-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/fullscreen-interactions-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/lock-basic-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/lock-sandboxed-iframe-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/lock-unlock-check-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/nested-documents-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/onchange-event-subframe-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/orientation-reading-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/unlock-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/256145@main">https://commits.webkit.org/256145@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a92763e44362aeb2076fb5ac8a985a34e2262bee

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94865 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4015 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27774 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104471 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/164734 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4100 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/32197 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/87168 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100394 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100534 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/2961 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/81303 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29946 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/84879 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/84428 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/72830 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38607 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18261 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36442 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/19543 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40364 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/42220 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2025 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42340 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38774 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

youennf commented on LayoutTests/imported/w3c/web-platform-tests/screen-orientation/active-lock-expected.txt:
```diff
@@ -1,5 +1,7 @@
 
-FAIL When performing a fragment navigation, the change must not abort promise_test: Unhandled rejection with value: object "NotSupportedError: Screen orientation locking is not supported"
+FAIL Performing a fragment navigation must not abort the screen orientation change promise_test: Unhandled rejection with value: object "NotSupportedError: Screen orientation locking is not supported"
+NOTRUN Performing a fragment navigation within an iframe must not abort the lock promise
```
Strange, usually we have Timeouts there before NOTRUN.
Is it possible the cleanup step is causing this?

marcoscaceres replied:
Yeah, I didn't realize that `.exitFullscreen()` could throw. 

I'll patch that up here and on WPT.

marcoscaceres replied:
Sent https://github.com/web-platform-tests/wpt/pull/36708

youennf commented on LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-basic-expected.txt:
```diff
@@ -1,8 +1,7 @@
-PASS Test that screen.orientation.lock returns a pending promise.
-FAIL Test that screen.orientation.lock() is actually async promise_test: Unhandled rejection with value: object "NotSupportedError: Screen orientation locking is not supported"
+NOTRUN Test that screen.orientation.lock returns a pending promise.
+NOTRUN Test that screen.orientation.lock() is actually async
```
Ditto here, are these tests taking a long time to run. If so, we might want to skip them.

marcoscaceres replied:
I think this one is ok... but I'll recheck that it doesn't hang.

marcoscaceres replied:
Yeah, this one is fine (they run quickly).

youennf commented on LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/onchange-event-subframe-expected.txt:
```diff
@@ -1,3 +1,7 @@
+Harness Error (TIMEOUT), message = null
+
 PASS Test subframes receive orientation change events
+TIMEOUT Check directly that events are fired in right order (from top to bottom) Test timed out
```
Usually we prefer skipping timing out tests.

marcoscaceres replied:
Added skip to TestExpectations.

youennf reviewed (approved):
Can you check whether these tests are timing out and if so skip them until they no longer time out?

marcoscaceres committed:
Resync web-platform-tests/screen-orientation from upstream
https://bugs.webkit.org/show_bug.cgi?id=247111
rdar://101619487

Reviewed by Youenn Fablet.

* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/active-lock-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/active-lock.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/event-before-promise-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/event-before-promise.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/fullscreen-interactions-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/fullscreen-interactions.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-basic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-basic.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-sandboxed-iframe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-sandboxed-iframe.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-unlock-check-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/lock-unlock-check.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/nested-documents-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/nested-documents.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/non-fully-active.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/onchange-event-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/onchange-event-subframe-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/onchange-event-subframe.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/onchange-event.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/orientation-reading-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/orientation-reading.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/resources/orientation-utils.js:
(export.async attachIframe):
(export.async loadIframe): Deleted.
(export.getOppositeOrientation): Deleted.
(export.makeCleanup.return.async if): Deleted.
(export.makeCleanup): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/resources/sandboxed-iframe-locking.html:
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/unlock-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/screen-orientation/unlock.html: Added.
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/active-lock-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/fullscreen-interactions-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/lock-basic-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/lock-sandboxed-iframe-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/lock-unlock-check-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/nested-documents-expected.txt: Added.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/onchange-event-subframe-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/orientation-reading-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/screen-orientation/unlock-expected.txt: Added.

Canonical link: https://commits.webkit.org/256145@main
